vault/diagnose/tls.go

```diff
@@ -4,24 +4,52 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
+	"io/ioutil"
 
 	"github.com/hashicorp/vault/internalshared/listenerutil"
+	"github.com/hashicorp/vault/sdk/helper/tlsutil"
 	"github.com/hashicorp/vault/vault"
 )
 
+const minVersionError = "'tls_min_version' value %q not supported, please specify one of [tls10,tls11,tls12,tls13]"
+const maxVersionError = "'tls_max_version' value %q not supported, please specify one of [tls10,tls11,tls12,tls13]"
+
 func ListenerChecks(listeners []listenerutil.Listener) error {
 	for _, listener := range listeners {
 		l := listener.Config
-		err := TLSFileChecks(l.TLSCertFile, l.TLSKeyFile)
+		var err error
+
+		// Perform the TLS version check for listeners
+		if l.TLSMinVersion == "" {
+			l.TLSMinVersion = "tls12"
+		}
+		if l.TLSMaxVersion == "" {
+			l.TLSMaxVersion = "tls13"
+		}
+		_, ok := tlsutil.TLSLookup[l.TLSMinVersion]
+		if !ok {
+			return fmt.Errorf(minVersionError, l.TLSMinVersion)
+		}
+		_, ok = tlsutil.TLSLookup[l.TLSMaxVersion]
+		if !ok {
+			return fmt.Errorf(maxVersionError, l.TLSMaxVersion)
+		}
+
+		// Perform checks on the TLS Cryptographic Information.
+		if l.TLSRequireAndVerifyClientCert {
```

Comment:
This is a mistake on my part -- this flag checks whether the listener verifies client certificates, but I'm using it to decide whether a root CA cert for the server, or intermediates, were passed in. I'm not sure how to split out the root CAs and inters from one single PEM file though.

Reply:
Fixed.